src/Microsoft.Azure.WebJobs.ServiceBus/ServiceBusAccount.cs

```diff
@@ -1,12 +1,59 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.WebJobs.Host;
+using System;
+using System.Globalization;
 
 namespace Microsoft.Azure.WebJobs.ServiceBus
 {
     internal class ServiceBusAccount
     {
-        public MessagingFactory MessagingFactory { get; set; }
+        private ServiceBusConfiguration _config;
+        private IConnectionProvider _connectionProvider;
+        private string _connectionString;
+
+
+        public ServiceBusAccount(ServiceBusConfiguration config, IConnectionProvider connectionProvider = null)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException("config");
+            }
+
+            _config = config;
+            _connectionProvider = connectionProvider;
+        }
+
+        internal ServiceBusAccount()
+        {
+        }
+
+        public virtual string ConnectionString
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_connectionString))
+                {
+                    _connectionString = _config.ConnectionString;
+                    if (_connectionProvider != null && !string.IsNullOrEmpty(_connectionProvider.Connection))
+                    {
+                        _connectionString = AmbientConnectionStringProvider.Instance.GetConnectionString(_connectionProvider.Connection);
+                    }
+
+                    if (string.IsNullOrEmpty(_connectionString))
+                    {
+                        throw new InvalidOperationException(
+                            string.Format(CultureInfo.InvariantCulture, "Microsoft Azure WebJobs SDK ServiceBus connection string '{0}{1}' is missing or empty.",
+                            "AzureWebJobs", _connectionProvider.Connection ?? ConnectionStringNames.ServiceBus));
```

Comment:
We need to be careful when throwing exceptions that include user-provided strings. Especially in this case, since the `ConnectionString` property is confusing and people may use their actual credentials. I added a utility to run these strings through -- look at the Sanitizer class.